### PR TITLE
Adjust RP deployment for MISE SideCar Container

### PR DESCRIPTION
**Pull request by SrinivasAtmakuri**
Deploys MISE Container on the RP VMSS for Prod, INT Environments. Will have another RP for RP-Config

### Which issue this PR addresses:

Fixes

### CODE REVIEW

**Comment by SrinivasAtmakuri** [comment]
// closing in favour of a different approach, do not want to create so many variables and instead pass the JSON directly.

File: pkg/deploy/assets/rp-production-parameters.json [hunk 1]
```diff
@@ -108,6 +108,23 @@
...
+            "value": "*"
+        },
+        "miseValidAudiences": {
+            "value": [
+                "https://management.azure.com/"
+            ]
+        },
+        "miseValidAppIDs": {
+            "value": []
+        },
```
Comment (bennerv) [inline pkg/deploy/assets/rp-production-parameters.json:127]: These will all get removed on `make generate`.  Update `templates_rp.go`, and you will need to specify default values for some of these.

File: pkg/deploy/generator/scripts/rpVMSS.sh [hunk 2]
```diff
@@ -123,9 +123,11 @@ mkdir -p /root/.docker
 REGISTRY_AUTH_FILE=/root/.docker/config.json az acr login --name "$(sed -e 's|.*/||' <<<"$ACRRESOURCEID")"
 
 MDMIMAGE="${RPIMAGE%%/*}/${MDMIMAGE##*/}"
+MISEIMAGE="${RPIMAGE%%/*}/${MDMIMAGE##*/}"
```
Comment (bennerv) [inline pkg/deploy/generator/scripts/rpVMSS.sh:126]: Referencing `MDMIMAGE` variable here, it should be `MISEIMAGE`

File: pkg/deploy/generator/scripts/rpVMSS.sh [hunk 3]
```diff
@@ -257,6 +259,93 @@ StartLimitInterval=0
...
+MISE_VALID_AUDIENCES='$MISE_VALID_AUDIENCES'
+MISE_VALID_APPIDs='$MISE_VALID_APPIDs'
+EOF
+
+cat >/app/mise/appsettings.json <<EOF
+{
+    "Version": "1",
+    "HeartbeatIntervalMs": 5000,
+    "AzureAd": {
+        "Instance": "https://login.microsoftonline.com/",
```
Comment (bennerv) [inline pkg/deploy/generator/scripts/rpVMSS.sh:281]: This isn't cloud aware.  This won't work in govcloud.

File: pkg/deploy/generator/scripts/rpVMSS.sh [hunk 4]
```diff
@@ -257,6 +259,93 @@ StartLimitInterval=0
...
+        "ClientId": "$AZURE_FP_CLIENT_ID",
+        "TenantId": "$AZURE_TENANT_ID",
+        "MinimumDataClassificationCategory": "SystemMetadata",
+        "InboundPolicies": [
+            {
+                "Label": "aro-policy",
+                "Authority": "https://login.microsoftonline.com/$AZURE_TENANT_ID/v2.0",
```
Comment (bennerv) [inline pkg/deploy/generator/scripts/rpVMSS.sh:288]: Same as above.  Won't work in govcloud

**Review by bennerv — CHANGES_REQUESTED** [review]

**Comment by SrinivasAtmakuri** [comment]
RP-Config: https://msazure.visualstudio.com/AzureRedHatOpenShift/_git/RP-Config/pullrequest/9901368

**Comment by SrinivasAtmakuri** [comment]
Code changes are moved to PR: https://github.com/Azure/ARO-RP/pull/3555

**Pull request closed without merging**